src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -130,10 +130,17 @@ def new_bootstrap_configuration(config_env, log_file_path, real_record_path, rec
                 'component': CompositeLogger,
                 'component_args': ['env_layer', 'file_logger'],
                 'component_kwargs': {
-                    'current_env': config_env
+                    'current_env': config_env,
+                    'telemetry_writer': None  # Has to be initialized without telemetry_writer to avoid running into a circular dependency loop. Telemetry writer within composite logger will be set later after telemetry writer has been initialized
+                }
+            },
+            'telemetry_writer': {
+                'component': TelemetryWriter,
+                'component_args': ['composite_logger', 'env_layer'],
```

Comment:
Re-order this to env_layer, composite_logger -- there is a call-order established for consistency across classes in this file. See lines 163, 168, etc. for example.